src/serial.rs

```diff
@@ -273,6 +460,9 @@ pub mod config {
                 stop_bits: StopBits::STOP1,
                 flow_control: FlowControl::None,
                 flow_control_rts_threshold: 122,
+                rx_buffer_size: UART_FIFO_SIZE * 2,
```

Comment:
in ESP-IDF the limit for RX and TX buffer sizes are each 128, which this (256) exceeds

(see in IDF, `uart.c: uart_driver_install()`:

```C
    ESP_RETURN_ON_FALSE((rx_buffer_size > SOC_UART_FIFO_LEN), ESP_FAIL, UART_TAG, "uart rx buffer length error");
    ESP_RETURN_ON_FALSE((tx_buffer_size > SOC_UART_FIFO_LEN) || (tx_buffer_size == 0), ESP_FAIL, UART_TAG, "uart tx buffer length error");
```
where `SOC_UART_FIFO_LEN` is 128 on all SOCs)

Reply:
Looking at the line just above:
```C
 ESP_RETURN_ON_FALSE((uart_num < UART_NUM_MAX), ESP_FAIL, UART_TAG, "uart_num error");
```
The first parameter logically except `true`; thus `rx_buffer_size` and `tx_buffer_size` are expected to be more than `SOC_UART_FIFO_LEN`  if I understand that correclty.
Also if you look a bit bellow in `uart_driver_install()`, you see that those buffer sizes are used to allocate memory for buffering between the UART FIFO and userspace.

Also, example from the Espressif doc: https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/uart.html#driver-installation
![image](https://user-images.githubusercontent.com/159235/178961408-12b46ba7-f7fa-4cfe-9427-31bab01cda41.png)